doc build fails with sphinx 4.2: deprecated API needs updating

xtensor docs previously built under sphinx 3.5.  I've now got sphinx 4.2 installed, which is making the docs fail to build:
```
$ cd docs; make html
...
sphinx-build -b html -d build/doctrees   source build/html
Running Sphinx v4.2.0
making output directory... done

Exception occurred:
  File "/projects/mathlibs/build/xtensor/docs/source/conf.py", line 19, in setup
    app.add_stylesheet("main_stylesheet.css")
AttributeError: 'Sphinx' object has no attribute 'add_stylesheet'
The full traceback has been saved in /tmp/sphinx-err-tqmfa670.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [Makefile:56: html] Error 2
```

The tmp file mentioned there reports much the same:
```
# Sphinx version: 4.2.0
# Python version: 3.9.7 (CPython)
# Docutils version: 0.16 release
# Jinja2 version: 3.0.1
# Last messages:

# Loaded extensions:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/sphinx/cmd/build.py", line 276, in build_main
    app = Sphinx(args.sourcedir, args.confdir, args.outputdir,
  File "/usr/lib/python3/dist-packages/sphinx/application.py", line 245, in __init__
    self.config.setup(self)
  File "/projects/mathlibs/build/xtensor/docs/source/conf.py", line 19, in setup
    app.add_stylesheet("main_stylesheet.css")
AttributeError: 'Sphinx' object has no attribute 'add_stylesheet'
```

https://github.com/sphinx-doc/sphinx/issues/7747 indicates that "the API was deprecated at 1.8.0. Please use app.add_css_file() instead."

--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -16,7 +16,7 @@ html_theme = "sphinx_rtd_theme"
 html_theme_path = [sphinx_rtd_theme.get_html_theme_path()]
 
 def setup(app):
-    app.add_stylesheet("main_stylesheet.css")
+    app.add_css_file("main_stylesheet.css")
 
 extensions = ['breathe']
 breathe_projects = { 'xtensor': '../xml' }

